recordedit/form.controller.js

```diff
@@ -484,6 +478,40 @@
             }
         }
 
+        function populateSubmissionRow(modelRow, submissionRow, originalTuple, columns, editOrCopy) {
+            var transformedRow = transformRowValues(modelRow);
+            columns.forEach(function (column) {
+                // If the column is a pseudo column, it needs to get the originating columns name for data submission
+                if (column.isPseudo) {
+
+                    var foreignKeyColumns = column.foreignKey.colset.columns;
+                    for (var k = 0; k < foreignKeyColumns.length; k++) {
+                        var referenceColumn = foreignKeyColumns[k];
+                        var foreignTableColumn = column.foreignKey.mapping.get(referenceColumn);
+                        // check if value is set in submission data yet
+                        if (!submissionRow[referenceColumn.name]) {
+                            /**
+                             * User didn't change the foreign key, copy the value over to the submission data with the proper column name
+                             * In the case of edit, the originating value is set on $rootScope.tuples.data. Use that value if the user didn't touch it (value could be null, which is fine, just means it was unset)
+                             * In the case of create, the value is unset if it is not present in submissionRows and because it's newly created it doesn't have a value to fallback to, so use null
+                            **/
+                            if (editOrCopy) {
+                                submissionRow[referenceColumn.name] = originalTuple.data[referenceColumn.name] || null;
```

Comment:
The comparison might be a problem if `originalTuple.data[referenceColumn.name]` was ever `0` or some other falsy value.